### ISSUE TITLE
2020-11-14-computing-in-the-cloud-with-google-and-rstudio

### ISSUE THREAD

**Issue by mkoohafkan** [mkoohafkan/mkoohafkan.github.io#45]
Comments on https://hydroecology.net/computing-in-the-cloud-with-google-and-rstudio/

**Comment by mwibbenmeyer** [comment]
Very helpful! Were you ever able to configure persistent storage on your VM, or did you have to load files from the the Google Storage bucket every time you logged in?

**Comment by mkoohafkan** [comment]
@mwibbenmeyer unfortunately not... I tried following the instructions in the [Persistent RStudio Server vignette](https://cran.r-project.org/web/packages/googleComputeEngineR/vignettes/persistent-rstudio.html) but didn't follow through. I was on a time crunch to complete my task so I stuck with the simpler approach. At the end of the day, I think the procedure in the vignette just automates file transfer to/from the bucket when you start/stop the VM so I'm not sure I'd *really* call it "persistent storage" in any case. The more interesting approach is to "mount" a storage bucket as a drive on the VM--which apparently you can do with [FUSE](https://cloud.google.com/storage/docs/gcs-fuse)--but I didn't dig into it... and now my free trial is over!

**Comment by mwibbenmeyer** [comment]
Yes, I was also following the instructions in that vignette, but ultimately had trouble getting it to work and ended up coming to the conclusion that automating transfer from the bucket was likely not worth the trouble. I haven't seen this FUSE solution, so I'll take a look at that. Apparently it is also possible to mount a persistent storage disk to VMs, as [here](https://cloud.google.com/compute/docs/disks/add-persistent-disk) but I wasn't fully able to make that work either. Ultimately I think I'll resort to transferring files back and forth to the bucket each session.
